webui/react/src/components/kit/Select.tsx

```diff
@@ -0,0 +1,159 @@
+  filterSort?: (a: LabeledValue, b: LabeledValue) => 1 | -1;
+  id?: string;
+  label?: string;
+  maxTagCount?: number | 'responsive';
+  maxTagPlaceholder?: string;
+  mode?: 'multiple' | 'tags';
+  onBlur?: () => void;
+  onChange?: (value: T, option: Options) => void;
+  onDeselect?: (selected: SelectValue, option: Options) => void;
+  onSearch?: (searchInput: string) => void;
+  onSelect?: (selected: SelectValue, option: Options) => void;
+  options?: LabeledValue[];
+  placeholder?: string;
+  placement?: 'bottomLeft' | 'bottomRight' | 'topLeft' | 'topRight';
+  ref?: React.Ref<RefSelectProps>;
+  showSearch?: boolean;
+  value?: T;
+}
+
+const countOptions = (children: React.ReactNode, options?: Options): number => {
+  let count = 0;
+
+  if (options) return options.length;
+  if (!children) return count;
+
+  if (Array.isArray(children)) {
+    count += children
+      .map((child) => countOptions(child, options))
+      .reduce((acc, count) => acc + count, 0);
+  }
+
+  const childType = (children as React.ReactElement).type;
+  const childProps = (children as React.ReactElement).props;
+  const childList = (childProps as React.ReactPortal)?.children;
+  if (childType === Option) count++;
+  if (childType === OptGroup && childList) count += countOptions(childList, options);
+
+  return count;
+};
+
+const Select: React.FC<React.PropsWithChildren<Props>> = forwardRef(function Select(
+  {
+    allowClear,
+    defaultValue,
+    disabled,
+    disableTags = false,
+    enableSearchFilter = true,
+    filterOption,
+    filterSort,
+    id,
+    label,
+    mode,
+    onChange,
+    onBlur,
+    onDeselect,
+    onSearch,
+    onSelect,
+    options,
+    placeholder,
+    placement,
+    showSearch,
+    value,
+    maxTagPlaceholder,
+    children,
+  }: React.PropsWithChildren<Props>,
+  ref?: React.Ref<RefSelectProps>,
+) {
+  const [isOpen, setIsOpen] = useState(false);
+  const classes = [css.base];
+
+  if (disableTags) classes.push(css.disableTags);
+  if (mode === 'multiple') {
+    classes.push(css.multiple);
+  }
+  const optionsCount = useMemo(() => countOptions(children, options), [children, options]);
+
+  const [maxTagCount, maxTagPlaceholderValue] = useMemo((): [
+    number | undefined | 'responsive',
+    string,
+  ] => {
+    if (!disableTags) return [undefined, maxTagPlaceholder ? maxTagPlaceholder : ''];
+    const count = Array.isArray(value) ? value.length : value ? 1 : 0;
+    const itemLabel = 'selected';
+    const placeholder = count === optionsCount ? 'All' : `${count} ${itemLabel}`;
+    return isOpen ? [0, ''] : [0, placeholder];
+  }, [disableTags, isOpen, optionsCount, maxTagPlaceholder, value]);
+
+  const handleDropdownVisibleChange = useCallback((open: boolean) => {
+    setIsOpen(open);
+  }, []);
+  const handleFilter = useCallback((search: string, option?: DefaultOptionType): boolean => {
+    let label: string | null = null;
+    if (option?.children) {
```

Comment:
nit: I'd replace this with `if (!option.children) return false` to reduce nesting if statements